dusty/systems/virtualbox/__init__.py

```diff
@@ -40,8 +40,40 @@ def _start_docker_vm():
     call_demoted(['VBoxManage', 'modifyvm', 'boot2docker-vm', '--memory', '{}'.format(memory_size)])
     check_call_demoted(['boot2docker', 'start'], redirect_stderr=True)
 
+def _stop_docker_vm():
+    """Stop the boot2docker VM if it is not already stopped."""
+    logging.info('Stopping the boot2docker VM')
+    check_call_demoted(['boot2docker', 'stop'], redirect_stderr=True)
+
+def _get_vm_config():
+    return check_output_demoted(['VBoxManage', 'showvminfo', '--machinereadable', 'boot2docker-vm']).splitlines()
+
+def _vm_is_using_virtio():
+    """Return boolean of whether the boot2docker VM has virtio configured on any of its
+    network interfaces."""
+    for line in _get_vm_config():
+        if 'nictype' in line and 'virtio' in line:
+            return True
+    return False
+
+def _apply_virtio_networking_fix():
+    """Modify all boot2docker VM network interfaces to use the preferred VM NIC type (generally
+    in place of virtio)."""
+    for idx, _ in enumerate(filter(lambda line: 'nictype' in line, _get_vm_config())):
+        logging.info('Setting VM NIC #{} to {}'.format(idx + 1, constants.VM_NIC_TYPE))
```

Comment:
I think you could theoretically set them up not starting at number 1, but that's not the default and I have no idea why anyone would do that.